core/src/main/scala/org/apache/spark/scheduler/BlacklistStrategy.scala

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+import scala.collection.mutable
+import org.apache.spark.Logging
+import org.apache.spark.SparkConf
+import org.apache.spark.util.SystemClock
+import org.apache.spark.util.Clock
+
+/**
+ * The interface to determine executor blacklist and node blacklist.
+ */
+private [scheduler] trait BlacklistStrategy {
+  /** Defined a time interval to expire failure information of executors */
+  val expireTimeInMilliseconds: Long
+
+  /** Return executors in blacklist which are related to given taskIndex */
+  def getExecutorBlacklist(
+      executorIdToFailureStatus: mutable.HashMap[String, FailureStatus],
+      atomTask: BlacklistAtomTask, clock: Clock): Set[String]
+
+  /** Return all nodes in blacklist */
+  def getNodeBlacklist(
+      executorIdToFailureStatus: mutable.HashMap[String, FailureStatus]): Set[String]
+
+  /**
+   * Default implementation to remove failure executors from HashMap based on given time period.
+   * The return value identity whether or not it updated anything
```

Comment:
Reword to:
Choose which executors (if any) should be removed from the blacklist.  Return true if any executors are removed from the blacklist, false otherwise.   The default implementation removes exectors from the blacklist after [[expireTimeInMillis]].